src/lib/compress.c

```diff
@@ -310,3 +310,30 @@ int compress_wait(struct compress *compress, int timeout_ms)
 	return compress->ops->wait(compress->data, timeout_ms);
 }
 
+int compress_get_supported_codecs_by_name(const char *name, unsigned int flags,
+		unsigned int *codecs, unsigned int size)
+{
+	struct compress *compress;
+	int ret = -1;
+
+	compress = calloc(1, sizeof(struct compress));
+	if (!compress)
+		return ret;
+
+	if ((name[0] == 'h') || (name[1] == 'w') || (name[2] == ':')) {
```

Comment:
check is name != NULL first and string size > 3 ?

Reply:
It's an undefined input so the crash is ok. The assert may be added. The size checking is not required, because the '\0' termination character is matched correctly.